### ISSUE TITLE
[BUG] Array without type

### ISSUE THREAD

**Issue by lifaon74** [Microsoft/TypeScript#18078]
**TypeScript Version:**  2.4.2 (same for 2.4.1)

**Code**

```ts
const promise: Promise<[]> = Promise.resolve([]);
// correct code => const promise: Promise<any[]> = Promise.resolve([]);
```

**Actual behavior:**
The above code is voluntary errored : The promise should return an array without type (the syntax is incorrect) and is of course invalid. When trying to compile, typescript crash without handling the exception. This result in a really difficult situation to find the syntax error. I took a few hours searching why typescript was crashing... because I didn't realize I forgot a type into one of my promises.

Crash both on typescript 2.4.2 and 2.4.1 (not tested on other versions).

**Expected behavior:**
This syntax error should be handled and noticed to the user.

**Comment by ikatyang** [comment]
Duplicate of #17548 and #17925, fixed by #17311 (v2.5).

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
